lib/types/contract.ts

```diff
@@ -0,0 +1,17 @@
+export interface Contract {
```

Comment:
Is this of the same format like the one we accept in the releases & the hostApps?
If not, then it might better be names like `DeviceTypeContract`.
Other than that LGTM (would be nice to have more accurate types than just `object` though). Feel free to self-certify.

Reply:
This is the same format for all contracts based on the current contract specification. This is also why some of the types are objects

Reply:
I'm not sure why this explains the why some props are `object`s, but we definitely don't need to block this PR any more.